### PR TITLE
[Merged by Bors] - chore(Topology): remove more bare open Classical

### PR DESCRIPTION
**Pull request by grunweg**
The only remaining occurrences are in Topology/Algebra.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/ContinuousFunction/Compact.lean [hunk 1]
```diff
@@ -26,12 +26,7 @@ you should restate it here. You can also use
 
 noncomputable section
 
-open scoped Classical
-open Topology NNReal BoundedContinuousFunction
-
-open Set Filter Metric
-
-open BoundedContinuousFunction
+open Topology NNReal BoundedContinuousFunction Set Filter Metric
```
Comment (YaelDillies) [inline Mathlib/Topology/ContinuousFunction/Compact.lean:29]: I think you're mangling what was originally `open_locale` with the bare `open`. Can you restore what was there before, but replacing one of the `open` with `open scoped`

File: Mathlib/Topology/LocallyConstant/Basic.lean [hunk 2]
```diff
@@ -19,15 +19,11 @@ This file sets up the theory of locally constant function from a topological spa
...
-
 -/
 
-
 variable {X Y Z α : Type*} [TopologicalSpace X]
 
-open Set Filter
-
-open Topology
+open Set Filter Topology
```
Comment (YaelDillies) [inline Mathlib/Topology/LocallyConstant/Basic.lean:26]: Same here

File: Mathlib/Topology/ContinuousFunction/Compact.lean (hunk 1 above)
Comment (grunweg) [reply Mathlib/Topology/ContinuousFunction/Compact.lean:29]: I am aware of this issue in general. In this case, replacing the `open` by `open scoped` errors.

**Comment by grunweg** [comment]
This is ready for review again.

**Review by YaelDillies — APPROVED** [review]
Thanks! 🚀

maintainer merge

**Pull request closed without merging**